tests/clickhouse-test

```diff
@@ -1212,7 +1212,26 @@ class TestCase:
             seconds_left = max(
                 args.timeout - (datetime.now() - start_time).total_seconds(), 20
             )
-            drop_database_query = "DROP DATABASE IF EXISTS " + database
+
+            leftover_tables = (
+                clickhouse_execute(
+                    args,
+                    f"SHOW TABLES FROM {database}",
+                    timeout=seconds_left,
+                    settings={
+                        "log_comment": args.testcase_basename,
+                    },
+                )
+                .decode()
+                .replace("\n", ", ")
+            )
+
+            if 0 != len(leftover_tables):
+                raise Exception(
+                    f"The test should cleanup its tables ({leftover_tables}), otherwise it is inconvenient for running it locally."
```

Comment:
Why? `clickhouse-test` creates (and drops) a separate database for each test even if you run it locally (unless you explicitly specified `--database`)

Reply:
+1 to this. Many test don't explicitly teardown their tables based on this fact too

Reply:
I run tests as follows:

```
clickhouse-client -n < test.sql
```
and
```
./test.sh
```

Reply:
Interesting. Out of curiosity, any reason not to use `./tests/clickhouse-test`?

Reply:
Slightly less direct feedback, plus the need to change directories.